30_regexp_lab/relab.txt

```diff
@@ -1,17 +1,7 @@
-Use grep to print only the lines from "input" for which:
-
-1. There is a binary number in the line
-
-2. There is an even binary number in the line
-
-3. Line includes a hexadecimal number
-
-4. Line starts with a lowercase letter
-
-5. Line is not empty
-
-6. Line starts with a lowercase letter and ends in a number
-
-7. Line is longer than 50 characters
-
-8. Line is of an even length
+1. grep -E '[0-9]{0,1}' input | head -11

```

Comment:
This one prints ALL the lines, even the ones without binary numbers. What it means is "print every line that has at least zero or one digit in it", but every line has at least 0 digits...

The following is the correct answer: `grep [01] input`.

Now this prints all lines that contain a 0 or 1 in them, which is nice but will also print a line that has the number 70 (which is not really a binary number).

We can improve with the `-w` option to grep, which means to search for full words. The command `egrep -w [01]+ input` will therefore print only sequences of zeroes and ones that form a full word